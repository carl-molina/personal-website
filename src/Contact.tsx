import React from "react";
import "./Contact.css";

/** Contact: presentational component for contact info.
 *
 * Props: none
 * State: none
 *
 * App -> RoutesList -> Contact
 */

function Contact() {

    return (
        <>
            <h5 className="Contact-email">carl@carlmolina.com</h5>
            <div className="Contact-icons">
                <h5><a href="https://www.linkedin.com/in/carlmolina/" className="nav-link" target="_blank" rel="noopener"><i className="fa-brands fa-linkedin"></i></a></h5>
                <h5><a href="https://github.com/carl-molina" className="nav-link" target="_blank" rel="noopener"><i className="fa-brands fa-github"></i></a></h5>
                <h5><a href="https://medium.com/@carlandrew.molina" className="nav-link" target="_blank" rel="noopener"><i className="fa-brands fa-medium"></i></a></h5>
                <h5><a href="https://twitter.com/carlandmolina" className='nav-link' target="_blank" rel="noopener"><i className="fa-brands fa-x-twitter"></i></a></h5>
                <h5><a href="mailto:carl@carlmolina.com" className="nav-link"><i className="fa-solid fa-envelope"></i></a></h5>
            </div>
        </>

    );
}

export default Contact;
