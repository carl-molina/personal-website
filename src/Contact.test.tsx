import React from "react";
import { test, expect, describe } from "vitest";
import { render, screen, fireEvent } from '@testing-library/react';

import Contact from './Contact';


describe("Projects smoke test", function () {
  test('renders without crashing', function () {
      render(<Contact />);
  });
});


describe("Projects snapshot test", function () {
  test('matches snapshot', function () {
      const { container } = render(<Contact />);
      expect(container).toMatchSnapshot();
  });
});
