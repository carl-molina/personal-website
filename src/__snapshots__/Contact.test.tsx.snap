// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact snapshot test > matches snapshot 1`] = `
<div>
  <div
    class="Contact-container"
  >
    <h5
      class="Contact-email"
    >
      carl@carlmolina.com
    </h5>
    <div
      class="Contact-icons"
    >
      <h5>
        <a
          class="nav-link"
          href="https://www.linkedin.com/in/carlmolina/"
          rel="noopener"
          target="_blank"
        >
          <i
            class="fa-brands fa-linkedin"
          />
        </a>
      </h5>
      <h5>
        <a
          class="nav-link"
          href="https://github.com/carl-molina"
          rel="noopener"
          target="_blank"
        >
          <i
            class="fa-brands fa-github"
          />
        </a>
      </h5>
      <h5>
        <a
          class="nav-link"
          href="https://medium.com/@carlandrew.molina"
          rel="noopener"
          target="_blank"
        >
          <i
            class="fa-brands fa-medium"
          />
        </a>
      </h5>
      <h5>
        <a
          class="nav-link"
          href="https://twitter.com/carlandmolina"
          rel="noopener"
          target="_blank"
        >
          <i
            class="fa-brands fa-x-twitter"
          />
        </a>
      </h5>
      <h5>
        <a
          class="nav-link"
          href="mailto:carl@carlmolina.com"
        >
          <i
            class="fa-solid fa-envelope"
          />
        </a>
      </h5>
    </div>
  </div>
</div>
`;
